### PR TITLE
unit tests for deviceType plugin deployer 

### PR DESCRIPTION
**Pull request by GDLMadushanka**
## Automation tests
 - Unit tests 
   > 12% coverage
 - Integration tests
   >  N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package org.wso2.carbon.device.mgt.extensions.device.type.deployer;
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java:19]: Add the license header

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.device.mgt.extensions.device.type.template.DeviceTypeConfigIdentifier;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DeviceTypePluginDeployerTest {
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java]: Add a class level comment

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    private DeviceTypePluginDeployer deviceTypePluginDeployer;
+    private DeploymentFileData deploymentFileData;
+
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java]: Shall we remove the extra line

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    private Field deviceTypeServiceRegistrations = null;
+    private Field deviceTypeConfigurationDataMap = null;
+    private ServiceRegistration serviceRegistration = null;
+
+    @BeforeClass
+    public void init() throws Exception {
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java]: Why are throwing generic exceptions? Is there any part which throw generic exception? If not shall we throw specific exceptions.

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        deviceTypePluginDeployer = Mockito.mock(DeviceTypePluginDeployer.class, Mockito.CALLS_REAL_METHODS);
+        serviceRegistration = Mockito.mock(ServiceRegistration.class, Mockito.CALLS_REAL_METHODS);
+        Mockito.doReturn(serviceRegistration).when(deviceTypePluginDeployer).registerDeviceType(Mockito.any(),
+                Mockito.any());
+
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java]: Remove extra lines. Plese do so in the following code segment as well

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/resources/android.xml [hunk 6]
```diff
@@ -0,0 +1,383 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+  ~ Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/resources/android.xml]: As we are adding as new files shall we change the license year

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/resources/testng.xml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/resources/testng.xml]: Remove this extra line

**Review by Megala21 — CHANGES_REQUESTED** [review]

**Comment by Megala21** [comment]
Since we are adding new test cases, in order to make this affect the overall test coverage, plese add the path of exec file to the test-coverage component.

**Review by Megala21 — APPROVED** [review]

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java (hunk 1 above)
Comment (GDLMadushanka) [reply components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java:19]: +1 Fixed

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/resources/testng.xml (hunk 7 above)
Comment (GDLMadushanka) [reply components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/resources/testng.xml]: +1 Fixed

**Pull request merged**